File: rules/go/tests/go.xss.direct-write/xss.go
package main

import (
	"fmt"
	"net/http"
	"text/template"
)

func direct(w http.ResponseWriter, r *http.Request) {
	w.Write([]byte(r.FormValue("q"))) // crit:expect go.xss.direct-write
}

func fprintf(w http.ResponseWriter, r *http.Request) {
	fmt.Fprintf(w, "<p>%s</p>", r.URL.Query().Get("q")) // crit:expect go.xss.direct-write
}

func escaped(w http.ResponseWriter, r *http.Request) {
	w.Write([]byte(template.HTMLEscapeString(r.FormValue("q")))) // crit:expect-not go.xss.direct-write
}

func static(w http.ResponseWriter) {
	w.Write([]byte("<p>hello</p>")) // crit:expect-not go.xss.direct-write
}

File: rules/csharp/tests/cs.xss.response-write/Xss.cs
class XssTests {
  void Direct() {
    var q = Request.QueryString["q"];
    Response.Write("<p>" + q + "</p>"); // crit:expect cs.xss.response-write
  }
  void ViaParams() {
    Response.Write(Request.Params["q"]); // crit:expect cs.xss.response-write
  }
  void Encoded() {
    var q = Request.QueryString["q"];
    Response.Write(HttpUtility.HtmlEncode(q)); // crit:expect-not cs.xss.response-write
  }
  void Static() {
    Response.Write("<p>hello</p>"); // crit:expect-not cs.xss.response-write
  }
}
